File: observability-kit-test/observability-kit-agent-it/src/main/java/com/vaadin/observability/test/MainView.java
package com.vaadin.observability.test;

import com.vaadin.flow.component.UI;
import com.vaadin.flow.component.dependency.JavaScript;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.H1;
import com.vaadin.flow.component.html.NativeButton;
import com.vaadin.flow.router.Route;

@Route("")
public class MainView extends Div {

    public MainView() {
        add(new H1("Observability Kit IT"));
        NativeButton clientSideError = new NativeButton("Client side errors",
                ev -> {
                    UI.getCurrent().getPage()
                            .addJavaScript("module_with_error.js");
                    UI.getCurrent().getPage().executeJs(
                            "invokeNotExistingFunctionFromServer();");
                });
        clientSideError.setId("clientSideError");
        add(clientSideError);
        getElement().executeJs("breakMe();");
    }
}

File: observability-kit-test/observability-kit-agent-hilla-it/frontend/views/main/main-view.ts
import {html, LitElement, TemplateResult} from 'lit';
import { customElement } from 'lit/decorators.js';

@customElement('main-view')
export class MainView extends LitElement {
  protected override createRenderRoot(): Element | ShadowRoot {
    return this;
  }

  protected override render(): TemplateResult {
    return html`
      <h1>Observability Kit Hilla IT</h1>
      <button id="clientSideError"
              @click="invokeNotExistingFunctionFromServer()">
        Client side errors
      </button>
    `;
  }

  invokeNotExistingFunctionFromServer() {

  }
}
